Rendering/WDE.MapSpawns/Rendering/SpawnContextMenu.cs

```diff
@@ -1,23 +1,111 @@
+using System;
 using System.Windows.Input;
 using Prism.Commands;
 using WDE.Common.Services;
+using WDE.Common.Services.MessageBox;
 using WDE.MapSpawns.Models;
 using WDE.MapSpawns.ViewModels;
+using WDE.SqlQueryGenerator;
+using WDE.DatabaseEditors.Services;
+using WDE.DatabaseEditors.Data.Interfaces;
 
 namespace WDE.MapSpawns.Rendering;
 
 public class SpawnContextMenu
 {
     private readonly ISpawnSelectionService spawnSelectionService;
-    
+    private readonly IDatabaseQueryExecutor mySqlExecutor;
+    private readonly ITableDefinitionProvider definitionProvider;
+    private readonly IMessageBoxService messageBoxService;
+
     private ICommand CopyGuidCommand { get; }
+    private ICommand CopyPositionXCommand { get; }
+    private ICommand CopyPositionYCommand { get; }
+    private ICommand CopyPositionZCommand { get; }
+    private ICommand CopyOrientationCommand { get; }
+    private ICommand UpdateValuesCommand { get; }
 
-    public SpawnContextMenu(ISpawnSelectionService spawnSelectionService,
-        IClipboardService clipboardService)
+    public SpawnContextMenu(
+        ISpawnSelectionService spawnSelectionService,
+        IClipboardService clipboardService,
+        ITableEditorPickerService tableEditorPickerService,
+        IDatabaseQueryExecutor mySqlExecutor,
+        ITableDefinitionProvider definitionProvider,
+        IMessageBoxService messageBoxService)
     {
         this.spawnSelectionService = spawnSelectionService;
+        this.mySqlExecutor = mySqlExecutor;
+        this.definitionProvider = definitionProvider;
+        this.messageBoxService = messageBoxService;
 
         CopyGuidCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.Guid.ToString()));
+        CopyPositionXCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.WorldObject!.Position.X.ToString()));
+        CopyPositionYCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.WorldObject!.Position.Y.ToString()));
+        CopyPositionZCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.WorldObject!.Position.Z.ToString()));
+        CopyOrientationCommand = new DelegateCommand<SpawnInstance>(inst =>
+        {
+            CreatureSpawnInstance? creature = inst as CreatureSpawnInstance;
+            GameObjectSpawnInstance? go = inst as GameObjectSpawnInstance;
+
+            if (creature != null) 
+            { 
+                clipboardService.SetText(creature.Creature!.Orientation.ToString()); 
+            }
+            else if (go != null) 
+            {
+                clipboardService.SetText(go.GameObject!.Orientation.ToString());
+            }
+        });
+        UpdateValuesCommand = new DelegateCommand<SpawnInstance>(async (inst) =>
+        {
+            var trans = Queries.BeginTransaction();
```

Comment:
Very good job on finding and using the syntax for writing SQL queries! :D

However, as you said, compatibility with different cores is a question. In case of creature/gameobject position I think the db structure is the same, but... there is a way of writing core-independent queries in the editor.

 There is no code for generating core-independent update position query, but let me show how it works for other query - updating quest chaining. 

```
IQueryGenerator<QuestChainDiff> queryGenerator; // get via constructor
var query = queryGenerator.Update(new QuestChainDiff() {
     Id = 1235,
     PrevQuestId = 1,
     NextQuestId = 2
});
mysqlExecutor.Execute(query);
```

how it works underneath: there are two implementations for `IUpdateQueryProvider<QuestChainDiff>`: `TrinityQuestQueryProvider` `MangosQuestQueryProvider`, which generate query like you generate it here.


So it would be great if you could create  `CreatureSpawnModelDiff` struct with an entry, guid, positions, orientations, create a `class CreatureUpdateQueryProvider : IUpdateQueryProvider<CreatureSpawnModelDiff>`, and in SpawnContextMenu use `IQueryGenerator<CreatureSpawnModelDiff>`. (note: there is already a `CreatureSpawnModelEssentials` struct which is used by query generator for insert and delete, but it is not really suitable for updating as its fields are not nullable

This way we can write integrations test for query generators (look at `DatabaseTester/QueryGeneratorTester.cs` for existing tests).

Reply:
I actually got the general gist of how to do it from ```TrinityQuestQueryProvider``` but didnt think that far that this of course also provides the opportunity to do a core-independant implementation. Thanks for pointing that out, I ll try adding that!

Reply:
Done! Hope its to your liking. As far as I can see cmangos uses the same column names as trinity [creature](https://github.com/cmangos/issues/wiki/creature) [gameobject](https://github.com/cmangos/issues/wiki/gameobject).